### PR TITLE
Encapsulate callable name access in messages.py

### PR DESCRIPTION
**Pull request by elazarg**
Trying to solve #4133 in two steps, in order to avoid conflicts on `semanal.py`. This step does not do much by itself - it replaces `callee.name` with `callable_name(callee)`. In later steps, the quotation marks can be added inside `callable_name` (which is a formatting function) instead of being part of `CallableType`.

As a side effect error messages are slightly more uniform, and `Function does not return a value` is replaced with `Expression evaluates to "None"` when the expression is not a function.

### CODE REVIEW

File: test-data/unit/check-expressions.test [hunk 1]
```diff
@@ -1525,9 +1525,7 @@ from typing import Iterator
 def f() -> Iterator[int]:
     yield 5
 def g() -> Iterator[int]:
-    a = yield from f()
-[out]
-main:5: error: Function does not return a value
+    a = yield from f()  # E: Expression evaluates to "None"
```
Comment (ilevkivskyi) [inline test-data/unit/check-expressions.test]: I don't think this message is better, it is still not clear what is the problematic expression. With previous wording, it may be better to add a function name.

Comment (elazarg) [reply test-data/unit/check-expressions.test]: But it's not necessarily a named function. I changed it to `Expression "yield from <iterator>" does not return a value`. Is that better?

File: mypy/messages.py [hunk 2]
```diff
@@ -656,61 +657,53 @@ def too_few_arguments(self, callee: CallableType, context: Context,
...
-        else:
-            self.fail('Function does not return a value', context)
+        name = callable_name(callee_type) or 'Function'
+        self.fail('{} does not return a value'.format(name), context)
+
+    def evaluates_to_none(self, context: Context) -> None:
+        self.fail('Expression evaluates to "None"', context)
+
+    def yield_from_iterator(self, context: Context) -> None:
+        self.fail('Expression "yield from <iterator>" does not return a value', context)
```
Comment (ilevkivskyi) [inline mypy/messages.py]: This part is not directly related to the PR as I understand. So that it is better to make a separate PR/issue.
These error messages deserve a separate discussion.

Comment (elazarg) [reply mypy/messages.py]: I thought I needed this change, but turns out you're right. Reverted.

(Although I do like the new messages better :) )

**Comment by ilevkivskyi** [comment]
@elazarg OK, I let us make the second step you mentioned.
